python/shotgun_model/shotgun_hierarchy_model.py

```diff
@@ -119,8 +134,141 @@ def item_from_path(self, path):
             :meth:`~shotgun-api3:shotgun_api3.Shotgun.nav_expand()` method.
         :returns: :class:`~PySide.QtGui.QStandardItem` or ``None`` if not found
         """
-        self._log_debug("Resolving model item for path %s" % path)
-        return self._ensure_item_loaded(path)
+        logger.debug("Resolving model item for path %s" % path)
+
+        if path == self._path:
+            return self.invisibleRootItem()
+        else:
+            return self._ensure_item_loaded(path)
+
+    class _NodeRefresher(QtCore.QObject):
```

Comment:
For me, classes within classes just makes the code more difficult to read. My opinion is that unless we have a good reason, we should keep them separate. I'm not super passionate about it though.